src/Microsoft.Extensions.OData.Migration/ExtensionMethods.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.OData.Edm;
+using System;
+using System.Globalization;
+
+namespace Microsoft.Extensions.OData.Migration
+{
+    /// <summary>
+    /// Contains extension methods for V3 IEdmType and System.Uri
+    /// </summary>
+    public static class ExtensionMethods
+    {
+        /// <summary>
+        /// (probably temporary until cleaner solution) method to concatenate a string to URI
+        /// </summary>
+        /// <param name="uri">base URI</param>
+        /// <param name="extra">string to append</param>
+        /// <returns>base URI with string appended</returns>
+        public static Uri Append(this Uri uri, string extra)

```

Comment:
Url append method needs to be done different (your comment is right). I don't think appending a URI like this works. If my uri is http://localhost/bob?d=bob then appending would add to the query portion. Are you just trying to add to the URL path?

Reply:
Replaced with built-in overload of Uri constructor (new Uri(basepath, relativestring))